client/aws/context.go

```diff
@@ -0,0 +1,18 @@
+package aws
+
+import (
+	"context"
+)
+
+type key string
+
+const clientsKey key = "clients"
+
+func NewContext(ctx context.Context, clients Clients) context.Context {
+	return context.WithValue(ctx, clientsKey, &clients)
+}
+
+func FromContext(ctx context.Context) (*Clients, bool) {
+	clients, ok := ctx.Value(clientsKey).(*Clients)
+	return clients, ok
+}
```

Comment:
I'd say it should rather sit in v10 package. `client` package doesn't have to know controller has to pass the client between resources.

Reply:
done